### PR TITLE
[Front-End] CSS for Detail Views of PO, SR, BB, and Genre

### PR DESCRIPTION
**Pull request by nex-pixel**
## Feature Description/Implementation
Added Delete Button
Edit Toggle is better
Add and Go back created
Disabled the Table when not modifying
Hide the delete line item button when not modifying
Added go back button to go back to list page respective to each
Fixed a Total revenue or cost bug not updating when deleting line items
Moved around some div elements and fixed unecessary and broken className stuff

## Dependencies
Affectst the Detail files of the ones in the title.

### CODE REVIEW

File: frontend/src/pages/detail/BBDetail.tsx [hunk 1]
```diff
@@ -188,8 +192,38 @@ export default function BBDetail() {
...
+
+  // Call to actually delete the element
+  const deleteBuyBackFinal = () => {
+    logger.debug("Delete Buy Back Finalized");
+    setDeletePopupVisible(false);
+    BUYBACK_API.deleteBuyBack({
+      id: id!,
+    })
+      .then(() => {
+        showSuccess(toast, "Buy Back Sale Deleted");
```
Comment (caseyszilagyi) [inline frontend/src/pages/detail/BBDetail.tsx:212]: I think we can change this to "Book Buyback Deleted/Book Buyback failed to delete". This goes for all instances of Buyback, he has it as one word in the requirements and refers to it as a "Book Buyback" so I think we should do the same thing.

File: frontend/src/pages/detail/BBDetail.tsx [hunk 2]
```diff
@@ -188,8 +192,38 @@ export default function BBDetail() {
...
+        navigate("/buy-backs");
+      })
+      .catch(() => showFailure(toast, "Buy Back Sale Failed to Delete"));
+  };
+
+  // The delete popup
+  const deletePopup = (
+    <DeletePopup
+      deleteItemIdentifier={" this buy back"}
```
Comment (caseyszilagyi) [inline frontend/src/pages/detail/BBDetail.tsx:221]: same here, can make buyback one word and call it "this book buyback"

File: frontend/src/pages/detail/BBDetail.tsx [hunk 3]
```diff
@@ -239,7 +274,10 @@ export default function BBDetail() {
       buybacks: apiSales,
     } as AddBBReq;
     BUYBACK_API.addBuyBack(buyBack)
-      .then(() => showSuccess(toast, "Buy back added successfully"))
+      .then(() => {
+        showSuccess(toast, "Buy back added successfully");
+        isGoBackActive ? navigate("/buy-backs") : window.location.reload();
```
Comment (caseyszilagyi) [inline frontend/src/pages/detail/BBDetail.tsx:279]: maybe change the url to book-buybacks as well

File: frontend/src/pages/detail/BBDetail.tsx [hunk 4]
```diff
@@ -360,6 +401,24 @@ export default function BBDetail() {
             className="p-button-success p-button-raised"
           />
         )}
+        {isModifiable && isBBAddPage && (
+          <ConfirmPopup
+            isVisible={isConfirmationPopupVisible}
+            hideFunc={() => setIsConfirmationPopupVisible(false)}
+            acceptFunc={onSubmit}
+            rejectFunc={() => {
+              console.log("reject");
```
Comment (caseyszilagyi) [inline frontend/src/pages/detail/BBDetail.tsx:410]: remove this log

File: frontend/src/pages/detail/BBDetail.tsx [hunk 5]
```diff
@@ -406,26 +465,57 @@ export default function BBDetail() {
...
+              icon="pi pi-arrow-left"
+              onClick={() => navigate("/buy-backs")}
+              className="p-button-sm my-auto ml-1"
+            />
+          </div>
+          <div className="pt-2 col-10">
             {isBBAddPage ? (
               <h1 className="p-component p-text-secondary text-5xl text-center text-900 color: var(--surface-800);">
-                Add Buy Back Sales
+                Add Buy Back Sale
```
Comment (caseyszilagyi) [inline frontend/src/pages/detail/BBDetail.tsx:488]: Same here, I think this should be "Add Book Buyback", and "Modify Book Buyback" right below

File: frontend/src/pages/detail/BBDetail.tsx [hunk 6]
```diff
@@ -406,26 +465,57 @@ export default function BBDetail() {
...
+                Buy Back Sale Details
               </h1>
             )}
           </div>
+          <div className="flex col-1">
+            {!isBBAddPage && (
+              <Button
+                type="button"
+                label="Delete"
+                tooltip="Delete this sales reconciliation"
```
Comment (caseyszilagyi) [inline frontend/src/pages/detail/BBDetail.tsx:505]: As discussed over slack, I don't think we need tooltips. Our UI is pretty intuitive at this point, and there's really nothing else that the delete button could delete besides the sales reconciliation

File: frontend/src/pages/detail/GenreDetail.tsx [hunk 7]
```diff
@@ -37,36 +69,58 @@ export default function GenreDetail() {
...
+                Modify Genre
+              </h1>
+            ) : (
+              <h1 className="p-component p-text-secondary text-5xl text-center text-900 color: var(--surface-800);">
+                Genre Details
+              </h1>
+            )}
           </div>
+          <div className="flex col-1">
+            <Button
```
Comment (caseyszilagyi) [inline frontend/src/pages/detail/GenreDetail.tsx:105]: Deleting should only be enabled if this genre has no books

**Review by caseyszilagyi — CHANGES_REQUESTED** [review]

File: frontend/src/pages/detail/GenreDetail.tsx (hunk 7 above)
Comment (caseyszilagyi) [reply frontend/src/pages/detail/GenreDetail.tsx:105]: Genre has numBooks or something that describes this

Comment (nex-pixel) [reply frontend/src/pages/detail/GenreDetail.tsx:105]: good call right now book_cnt return for genre detail is null for some reason

**Review by caseyszilagyi — APPROVED** [review]
LTGM

**Pull request merged**